bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+	Rewrites map[string]string `yaml:"rewrite" json:"rewrite,omitempty"`
+}
+
+// RegistryConfig contains configuration used to communicate with the registry.
+type RegistryConfig struct {
+	// Auth si a reference to a Secret containing information to authenticate to the registry.
+	// The Secret must provite a username attribute and a password attribute
+	//+optional
+	AuthSecret SecretRef `json:"authSecret,omitempty"`
+	// TLS is a pair of CA/Cert/Key which then are used when creating the transport
+	// that communicates with the registry.
+	//+optional
+	TLS TLSConfig `yaml:"tls" json:"tls,omitempty"`
+}
+
+// AuthConfig contains the config related to authentication to a specific registry
+// This is not used in the CRDs but is used for marshalling the registries.yaml file
+type AuthConfig struct {
+	// Username is the username to login the registry.
+	Username string `yaml:"username" json:"username,omitempty"`
+	// Password is the password to login the registry.
+	Password string `yaml:"password" json:"password,omitempty"`
+
+	// IdentityToken is used to authenticate the user and get
+	// an access token for the registry.
+	IdentityToken string `yaml:"identity_token" json:"identityToken,omitempty"`
+}
+
+// TODO: Beware here: TLSConfig is part of the Registry struct which has a double usage:
+// - Used in the RKE2Config CRD
+// - Used to generate a registries.yaml file for the RKE2 command line
+// In the CRD, CASecret attribute is a reference to a Secret in the Kube API, while in the registries.yaml CASecret is marshalled into ca_file which a file path on the system
+// This means the Registry object will need to be copied and processed before marshalling
+
+// TLSConfig contains the CA/Cert/Key used for a registry
+type TLSConfig struct {
+
+	// CASecret is a reference to a secret containing the Certificate Authority used to verify the registry's certificate
+	//+optional
+	CASecret SecretRef `json:"caSecret,omitempty"`
+
+	// CertSecret is a reference to a secret containing the Certificate used to authenticate to the registry
+	//+optional
+	CertSecret SecretRef `json:"certSecret,omitempty"`
+
+	// KeySecret is a reference to a secret containing the Key for the certificate used to authenticate to the registry
+	//+optional
+	KeySecret SecretRef `json:"keySecret,omitempty"`
+
+	//InsecureSkipVerify may be set to true to skip verifying the registry's certificate
+	//+optional
+	InsecureSkipVerify bool `yaml:"insecure_skip_verify" json:"insecureSkipVerify,omitempty"`
+}
+
+type EtcdS3 struct {
+
+	// Endpoint S3 endpoint url (default: "s3.amazonaws.com")
+	// +required
+	Endpoint string `json:"endpoint"`
+
+	// EndpointCA references the Secret that contains a custom CA that should be trusted to connect to S3 endpoint
+	// +optional
+	EndpointCA SecretRef `json:"endpointCA,omitempty"`
+
+	// SkipSslVerify Disables S3 SSL certificate validation
+	// +optional
+	SkipSslVerify string `json:"skipSslVerify,omitempty"`
```

Comment:
is this supposed to be a bool flag?

Reply:
As of RKE2's documentation, you are right, this should be a `bool`. However, does the same idea as the above comment for `InsecureSkipVerify` also apply here?

Reply:
Done, using `SSLVerifyPolicy` as a reusable enum type for both S3 backups and Private Registry configrations.